File: android/src/main/java/io/t6x/dust/capacitor/core/CorePlugin.kt
package io.t6x.dust.capacitor.core

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin
import io.t6x.dust.core.DustCore

@CapacitorPlugin(name = "Core")
class CorePlugin : Plugin() {

    companion object {
        const val CONTRACT_VERSION = DustCore.VERSION
    }

    @PluginMethod
    fun getContractVersion(call: PluginCall) {
        val result = JSObject()
        result.put("version", CONTRACT_VERSION)
        call.resolve(result)
    }
}

File: ios/Sources/CorePlugin/CorePlugin.swift
import Foundation
import Capacitor
@_exported import DustCore

@objc(CorePlugin)
public class CorePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CorePlugin"
    public let jsName = "Core"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getContractVersion", returnType: CAPPluginReturnPromise),
    ]

    private static let contractVersion = DustCoreVersion.current

    @objc func getContractVersion(_ call: CAPPluginCall) {
        call.resolve(["version": Self.contractVersion])
    }
}
